clang/docs/checkedc/Bounds-Widening-for-Null-Terminated-Arrays.md

```diff
@@ -0,0 +1,159 @@
+# Bounds Widening for Null-terminated Arrays
+
+## Null-terminated Arrays
+A null-terminated array is a sequence of elements in memory that ends with a
+null terminator. Checked C adds the type `nt_array_ptr<T>` to represent
+pointers to these kinds of arrays. These arrays can be divided into two parts:
+a prefix with bounds followed by a null terminator.
+
+An important property of a null-terminated array is that an element of the
```

Comment:
I don't think this is the right description of the properties of a null-terminated array.  I think it is better to describe the array as a prefix with bounds followed by a sequence of elements, ending with an element that is the null terminator.

Reply:
David, I have updated the description of the properties of a null-terminated array.